students/MikeSchincariol/session7/html_render.py

```diff
@@ -0,0 +1,200 @@
+import io
+
+class Element(object):
+
+    @classmethod
+    def check_content_is_compat(cls, content):
+        if isinstance(content, str) or isinstance(content, Element):
+            return True
+        else:
+            return False
+
+
+    def __init__(self, content=None, **kwargs):
+        '''
+        :param content: The string or element instance to initialize the element with.
+        :param kwargs: Attributes that are associated with the content
+        :return: Nothing
+        '''
+        self.tag = 'tag'
+        self.data = []
+        self.attr = kwargs
+
+        if content is not None:
+            if Element.check_content_is_compat(content):
+                self.data.append(content)
+            else:
+                raise ValueError("The 'content' argument must be a string or another Element")
+
+
+    def append(self, content, **kwargs):
+        '''
+        :param content: The string or element instance to add to the items already in the element
+        :param kwargs: Attributes that are associated with the content
+        :return: Nothing
+        '''
+
+        self.attr.update(kwargs)
+        if content is not None:
+            if Element.check_content_is_compat(content):
+                self.data.append(content)
+            else:
+                raise ValueError("The 'content' argument must be a string or another Element")
+
+
+    def _render_start_tag(self, file_out, ind, inline=True, self_closing=False):
+        try:
+            if not inline:
+                file_out.write("\n{0}".format(ind))
+            file_out.write("<{0}".format(self.tag))
+            for k,v in self.attr.items():
+                file_out.write(' {0}="{1}"'.format(k, v))
+            if self_closing:
+                file_out.write("/>")
+            else:
+                file_out.write(">")
+        except AttributeError:
+            raise ValueError("The 'file_out' argument must be a class or subclass of io.TextIOBase")
+
+
+    def _render_data(self, file_out, ind, inline=True):
```

Comment:
Can you think of a way to combine these three functions into one? Would that be more, or less, efficient?